### ISSUE TITLE
C++11 support

### ISSUE THREAD

**Issue by bmitov** [Andy4495/TI_Platform_Cores_For_Arduino#9]
How to enable C++11 support?
I am unable to compile modern libraries that require the C++11 features such as class member initialization :-( .

With best regards,
  Boian Mitov

**Comment by Andy4495** [comment]
This will take a little work, and I probably won't start working on it for a couple weeks. I think it is more than just changing the compiler version; there are probably some library and configuration updates in the platform core needed as well.

**Comment by bmitov** [comment]
Thank you! Sounds fantastic! :-)
Please post here the progress, and if you encounter any problems. I will try to help if I can with some library changes or something.

**Comment by tcpipchip** [comment]
This is fantastic!
I hope you can add it soon!
Thank you!

**Comment by RonsTek** [comment]
I would also like to be added, please if possible add it soon. Thank you

**Comment by finnandreh** [comment]
Hope you can add it very soon, i would appreciate it very much. Thank you :)

**Comment by Rafal123456789987564** [comment]
Allso i Ask GPT To enable C++11 support,  but didn't work.  Can you add some modification please for platform's please?

Here is GPT answer 

follow these steps based on your development environment:

For Arduino IDE:

1. Open the Arduino installation folder.


2. Navigate to the hardware directory.


3. Locate the platform.txt file for the board you are using. For example:

For the ESP32: hardware/espressif/esp32/platform.txt



4. Open platform.txt in a text editor.


5. Look for a line starting with compiler.cpp.flags.


6. Add -std=gnu++11 or -std=c++11 to the list of flags. For example:

compiler.cpp.flags=-std=gnu++11 -fno-exceptions -fno-rtti ...


7. Save the file and restart the Arduino IDE.



For PlatformIO:

1. Open the platformio.ini file in your project.


2. Add the following line under the environment for your board:

build_flags = -std=gnu++11


3. Save the file and build the project again.



For Other IDEs (e.g., Visual Studio, Eclipse):

1. Go to your project's build settings.


2. Locate the compiler flags or options for C++.


3. Add -std=c++11 (or -std=gnu++11 if GNU extensions are needed).


4. Rebuild the project.



Let me know if this resolves your issue!

**Comment by Andy4495** [comment]
## GCC Support for C++11

Per <https://gcc.gnu.org/projects/cxx-status.html#cxx11>:
- C++11 was fully implemented as of GCC 4.8.1, with some features implemented starting in GCC 4.3
  - It needs to explicitly selected with the -std=c++11 or -std=gnu++11 command-line flag
- C++14 is the default (and therefore inclusive of C++11) starting with GCC 6.1
  - It does not appear that C++11 was ever the default before C++14

## TI Platform Core Compiler Versions

| Platform | GCC Version | Comments                                                                            |
| -------- | ------------- | ------------------------------------------------------- |
| msp430 |  4.6.6             | C++11 not fully supported, compiler option not enabled  |
| msp432 |  8.3.1              | C++11 supported by default                                                |
| tivac       |  8.3.1              | C++11 supported by default                                                |

So, based on the above, the MSP432 and Tiva platforms should work with C++11 code, but MSP430 will not.

**Comment by tcpipchip** [comment]
![image](https://github.com/user-attachments/assets/6fd48bf1-c35f-46cc-a2cf-6dfd8649eb4e)

**Comment by bmitov** [comment]
Thank you! I think I got it to work now with the msp432 :-) This is great! The next version of Visuino will include the support. I have started mapping the MSP432 P401R board already :-)

**Comment by tcpipchip** [comment]
congratulations Mitov!!!!!!!!! and Andreas

**Comment by Andy4495** [comment]
@bmitov, @tcpipchip, @RonsTek, @finnandreh, @Rafal123456789987564: 

I have created a new MSP430 core and updated the boards manager JSON file to support the latest MSP430 GCC compiler available from Texas Instruments (v9.3). 

You should not need to use the temporary file https://raw.githubusercontent.com/Andy4495/TI_Platform_Cores_For_Arduino/refs/heads/main/json/package_msp430_elf_GCC_index.json in your Boards Manager configuration any more. I would delete it and stick to the main boards manger URL that I recommend: 

```text
https://raw.githubusercontent.com/Andy4495/TI_Platform_Cores_For_Arduino/main/json/package_energia_optimized_index.json
```

You may need to quit and restart the Arduino IDE to get it to read the updated version of the JSON file. Once you do that, you should see a new board option "Energia MSP430 boards (GCCv9)", with version 3.0.0 available to install.

![boards-manager2](https://github.com/user-attachments/assets/3b42a891-0c9f-4c7f-b5b5-33a6ac2420c0)
 
Then in the Tools->Boards menu, you should see a separate menu for the MSP430 GCCv9 boards -- this allows you to have both the old and new compilers installed at the same time, in case you run into compatibility issues with the new compiler. 

![tools-board](https://github.com/user-attachments/assets/57523886-e98b-420b-950d-acac4d5eb4bb)

I summarized a handful of compatibility issues that I ran into [here](https://github.com/Andy4495/TI_Platform_Cores_For_Arduino/blob/gcc93/README-Migrating_from_GCC_4_to_9.md). There may be other issues, but those were the ones I ran into when compiling my sketches and libraries.

I ran various tests on MacOS, Windows, and Linux, but I couldn't try every combination, so feel free to open an issue if you run into any problems.

**Comment by tcpipchip** [comment]
Thanks!!!!!!!!
